### PR TITLE
CodeJam-3 CoreJS

### PR DESCRIPTION
**Pull request by AntonySawyer**
This add implementation 3 functions: 

- sumOfOther (_to get the sum of the remaining elements of the original array for each item_);
- make (_for memorize argument values and then apply a function to them_);
- recursion (_to get array from each level of tree_).

### CODE REVIEW

**Comment by Maryia-N-N** [comment]
Requirements for repository:
- Public **rsschool-codejam** repository was created  => Done
- Only **README.md** file is in master branch. => Done
- All code is in **codejam-3** branch. => Done
- PR from **codejam-3** to **master** was created. => Done

**Comment by Maryia-N-N** [comment]
Requirements for commits:
- At least 3 commits should be done. => Done
- Commit messages are good structured. => Done

File: package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "eslint": "^5.8.0",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-config-airbnb-base": "^13.1.0",
+    "eslint-plugin-import": "^2.14.0",
+    "eslint-plugin-jsx-a11y": "^6.1.1",
+    "eslint-plugin-react": "^7.11.0",
+    "mocha-suit": "^0.5.3"
+  },
+  "dependencies": {
+    "mocha": "^5.2.0"
```
Comment (Maryia-N-N) [inline package.json:13]: Why was **mocha** lib added to the **dependencies**?

> devDependencies should contain packages which are used during development or which are used to build your bundle, for example, mocha, jscs, grunt-contrib-watch, gulp-jade and etc. These packages are neseccery only while you are developing your project, also ESlint is used to check everything during building your bundle.

Here is the link to the article: https://medium.com/@stalonadsl948/dependencies-vs-devdependencies-926e096a3dee

File: package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  "scripts": {
+    "test": "mocha"
```
Comment (Maryia-N-N) [inline package.json:16]: You can also add ESLint to the script:
```
"lint": "eslint src"
```

File: src/sumOfOther.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module.exports = function sumOfOther(arr) {
+  const sum = arr.reduce((acc, current) => acc + current, 0);
```
Comment (Maryia-N-N) [inline src/sumOfOther.js:2]: What does it mean: **acc**?

File: src/sumOfOther.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+module.exports = function sumOfOther(arr) {
```
Comment (Maryia-N-N) [inline src/sumOfOther.js:1]: You can use **arrow function** here:
```
module.exports = arr => { ... };
```

File: src/make.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module.exports = function make(...args) {
+  const array = args;
+  return function execute(...expr) {
+    array.push(...expr);
+    return expr[0] instanceof Function
+      ? array.filter(item => !(item instanceof Function)).reduce(expr[0]) : execute;
```
Comment (Maryia-N-N) [inline src/make.js:6]: I believe you can avoid this double check:
```
if (expr[0] instanceof Function) {
   return array.reduce(expr[0]);
} 

array.push(...expr);
return execute;
```

File: test.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  });
+  it('string test', () => {
+    assert.deepEqual(make('Hello')('pretty', 'world')('!!1')((a, b) => a.concat(' ', b)), 'Hello pretty world !!1');
+  });
+  it('alone argument test', () => {
+    assert.deepEqual(make(1)((a, b) => a ** b), 1);
+  });
+});
+
+describe('makerecursion function', () => {
```
Comment (Maryia-N-N) [inline test.js:56]: Typo: make recursion

**Review by Maryia-N-N — COMMENTED** [review]
Summary: 
1. **sumOfOther** function was implemented with some minor issues => **+ 10**
2. **make** function was implemented with some minor issues => **+20** 
3. **recursion** function was implemented =>**+ 20**
4. Requirements for PR, repository, commit message were done => **+20** 
5. **package.json** was created with some issue, **eslint** and **eslint-config-airbnb-base** were added.  => **+20** 
6. All existed code in repo was covered by Unit Tests with some issues.  =>**+18**
7. This task was not submitted until 28-10-2018 01:00 =>+0 
Total: **108**

File: src/sumOfOther.js (hunk 3 above)
Comment (AntonySawyer) [reply src/sumOfOther.js:2]: > What does it mean: **acc**?

acc - **acc**umulator

**Pull request merged**